src/hooks/useWalletConnection.ts

```diff
@@ -187,13 +201,22 @@ export const useWalletConnection = () => {
 
     if (isConnectedWagmi) await disconnectWagmi();
     if (isConnectedGraz) await disconnectGraz();
+    if (authenticated) await logout();
   }, [isConnectedGraz, isConnectedWagmi]);
```

Comment:
does `authenticated` need to be add to dep list?

Reply:
also: does `logout` need to be added? I see `login` is used as a dep in the `connectWallet` callback

Reply:
+1